### PR TITLE
Add redirects and tests

### LINKED ISSUE #638: Ensure redirects are handled correctly
A few cases that may not be handled correctly right now:
- user visits a group detail page they are not a member of
- user visits a page that does not exist at all (no route)
- item does not exist (e.g. group id that does not exist)
- other things like this (but for the store)

In general some of this stuff should/is be handled, but is maybe not completely correct right now. Should also have some tests unit and/or e2e.

### LINKED ISSUE #762: User page is empty if user hasn't joined any group
(more info needed?)

### PR DESCRIPTION
**Pull request by tiltec**
Closes #638 
Closes #762

I added some unit tests. I also addressed #762 because it was related: to work around race conditions when loading the page, I just try to load the entry directly from the server if it's not in vuex yet.

To do:
- [x] fix invalid group id redirect

### CODE REVIEW

File: src/store/modules/groups.spec.js [hunk 1]
```diff
@@ -116,7 +122,7 @@ describe('groups', () => {
     })
 
     it('can not join a group', async () => {
-      mockJoin.mockImplementationOnce(() => { throw new Error('some error') })
+      mockJoin.mockImplementationOnce(throws(Error('some error')))
```
Comment (nicksellen) [inline src/store/modules/groups.spec.js]: Ah, I didn't now `Error()` is the same as `new Error()`, looks neater, would be nice to move towards consistency. No change requested.

File: src/store/modules/groups.js [hunk 2]
```diff
@@ -95,10 +95,12 @@ export const actions = {
 
       commit(types.SET_ACTIVE, { groupId })
 
-      try {
-        await dispatch('fetchGroup', groupId)
-      }
-      catch (error) {
+      await dispatch('fetchGroup', groupId)
+      const hasError = Object.keys(getters['meta/status']('fetchGroup', groupId).validationErrors).length > 0
```
Comment (nicksellen) [inline src/store/modules/groups.js]: I have seen this elsewhere, perhaps we need an extra property return from the getter function, I would called it `hasValidationErrors` (to distinguish it from other errors).

File: src/store/modules/users.js [hunk 3]
```diff
@@ -248,5 +259,8 @@ export const mutations = {
   },
   [types.RECEIVE_USER] (state, { user }) {
     Vue.set(state.entries, user.id, user)
+    if (!state.idList.includes(user.id)) {
```
Comment (nicksellen) [inline src/store/modules/users.js]: Do we actually need/use the `idList` for users? I left it off when I added the agreements module as we only ever need to get by id.

Comment (tiltec) [reply src/store/modules/users.js]: Maybe for the users list? But they might need to get sorted on-the-fly anyways. I'd leave it in for now.. we could open an issue in the future.

File: src/store/modules/groups.spec.js (hunk 1 above)
Comment (tiltec) [reply src/store/modules/groups.spec.js]: It does work, but maybe it's not the best style? I remember reading that one should always use `new Error`.

File: src/store/modules/groups.js (hunk 2 above)
Comment (tiltec) [reply src/store/modules/groups.js]: Though about it, will add it!

**Pull request merged**